### PR TITLE
Avkhokhlov/light for stanalone

### PR DESCRIPTION
**Pull request by AvKhokhlov**

### CODE REVIEW

File: BaikalStandalone/Application/scene_load_utils.cpp [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+#include <scene_io.h>
+#include <image_io.h>
+#include <material_io.h>
+#include "XML/tinyxml2.h"
+
+using namespace Baikal;
+
+namespace
+{
+    void LoadMateirals(std::string const& basepath, Scene1::Ptr scene)
```
[the remark this answers is not in the document]
Comment (AvKhokhlov) [reply BaikalStandalone/Application/scene_load_utils.cpp]: Done

File: BaikalStandalone/Application/scene_load_utils.cpp [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+                auto image_io = ImageIo::CreateImageIo();
+                auto tex = image_io->LoadImage(elem->Attribute("tex"));
+                auto ibl = ImageBasedLight::Create();
+                ibl->SetTexture(tex);
+                ibl->SetMultiplier(elem->FloatAttribute("mul"));
+                light = ibl;
+            }
+            else
+            {
+                throw std::runtime_error("Unknown light type");
```
[the remark this answers is not in the document]
Comment (AvKhokhlov) [reply BaikalStandalone/Application/scene_load_utils.cpp]: Done

File: BaikalStandalone/Application/scene_load_utils.cpp [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+        auto root = doc.FirstChildElement("light_list");
+
+        if (!root)
+        {
+            throw std::runtime_error("Failed to open lights set file.");
+        }
+
+        tinyxml2::XMLElement* elem = root->FirstChildElement("light");
+        Light::Ptr light_instance;
```
[the remark this answers is not in the document]
Comment (AvKhokhlov) [reply BaikalStandalone/Application/scene_load_utils.cpp]: Yep

File: BaikalStandalone/Application/scene_load_utils.cpp [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+            elem = elem->NextSiblingElement("light");
+        }
+    }
+}
+
+
+Scene1::Ptr LoadScene(Baikal::AppSettings const& settings)
+{
+#ifdef WIN32
+    std::string basepath = settings.path + "\\";
```
[the remark this answers is not in the document]
Comment (AvKhokhlov) [reply BaikalStandalone/Application/scene_load_utils.cpp]: Probably

File: BaikalStandalone/Application/scene_load_utils.h [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+ ********************************************************************/
+
+#pragma once
+
+#include <memory>
+#include "app_utils.h"
+
+class Baikal::Scene1;
```
[the remark this answers is not in the document]
Comment (AvKhokhlov) [reply BaikalStandalone/Application/scene_load_utils.h]: Done

File: BaikalStandalone/Application/scene_load_utils.cpp [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+                light = spot;
+            }
+            else if (type == "direct")
+            {
+                light = DirectionalLight::Create();
+            }
+            else if (type == "ibl")
+            {
+                auto image_io = ImageIo::CreateImageIo();
+                auto tex = image_io->LoadImage(elem->Attribute("tex"));
```
[the remark this answers is not in the document]
Comment (AvKhokhlov) [reply BaikalStandalone/Application/scene_load_utils.cpp]: No, we can't

File: BaikalStandalone/Application/scene_load_utils.cpp [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+        auto mats = material_io->LoadMaterials(basepath + "materials.xml");
+        auto mapping = material_io->LoadMaterialMapping(basepath + "mapping.xml");
+        material_io->ReplaceSceneMaterials(*scene, *mats, mapping);
+    }
+
+    void LoadLights(std::string const& light_file, Scene1::Ptr scene)
+    {
+        if (light_file.empty())
+        {
+            return;
```
Comment (yozhijk) [inline BaikalStandalone/Application/scene_load_utils.cpp]: User probably does not intend to pass an empty light file, should that throw an exception?

File: BaikalStandalone/Application/scene_load_utils.cpp [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+        doc.LoadFile(light_file.c_str());
+
+        auto root = doc.FirstChildElement("light_list");
+
+        if (!root)
+        {
+            throw std::runtime_error("Failed to open lights set file.");
+        }
+
+        tinyxml2::XMLElement* elem = root->FirstChildElement("light");
```
Comment (yozhijk) [inline BaikalStandalone/Application/scene_load_utils.cpp]: auto?

File: BaikalStandalone/CMakeLists.txt [hunk 9]
```diff
@@ -14,7 +14,9 @@ set(APPLICATION_SOURCES
     Application/graph_scheme.h
     Application/graph_scheme.cpp
     Application/material_explorer.h
-    Application/material_explorer.cpp)
+    Application/material_explorer.cpp
+    Application/scene_load_utils.h
```
[the remark this answers is not in the document]
Comment (AvKhokhlov) [reply BaikalStandalone/CMakeLists.txt]: Nope

File: BaikalStandalone/Application/scene_load_utils.cpp (hunk 7 above)
Comment (AvKhokhlov) [reply BaikalStandalone/Application/scene_load_utils.cpp]: Nope, it's valid case, right there is no light file at all

File: BaikalStandalone/Application/scene_load_utils.cpp (hunk 8 above)
Comment (AvKhokhlov) [reply BaikalStandalone/Application/scene_load_utils.cpp]: Yep

**Pull request merged**